debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSnapshotChangeEventSource.java

```diff
@@ -401,15 +401,31 @@ protected void complete(SnapshotContext<MySqlPartition, MySqlOffsetContext> snap
     }
 
     /**
-     * Generate a valid sqlserver query string for the specified table
+     * Generate a valid MySQL query string for the specified table and columns
      *
      * @param tableId the table to generate a query for
      * @return a valid query string
      */
     @Override
     protected Optional<String> getSnapshotSelect(RelationalSnapshotContext<MySqlPartition, MySqlOffsetContext> snapshotContext,
-                                                 TableId tableId) {
-        return Optional.of(String.format("SELECT * FROM `%s`.`%s`", tableId.catalog(), tableId.table()));
+                                                 TableId tableId, List<String> columns) {
+        String snapshotSelectColumns = columns.stream()
+                .collect(Collectors.joining(", "));
+
+        return Optional.of(String.format("SELECT %s FROM `%s`.`%s`", snapshotSelectColumns, tableId.catalog(), tableId.table()));
+    }
+
+    @Override
+    protected String resolveColumnName(TableId tableId, String columnName) {
```

Comment:
@ani-sha, so the implementation of this method (here and elsewhere) is the last thing that gives me some headache:

* We don't quote the table/column name which potentially is problematic when using reserved words like "order" as identifiers
* The logic will fail to qualify with the table name if a column is named like the table

Interestingly, at least for MySQL the former doesn't seem to be actually a problem (see the test I added around a column named `order.select`, which succeeds without any quoting). But I'm not sure for others.

So I think what we should do is:

* get rid of the `columnName.contains(tableId.table())` check; there's exactly one call site for this method, and it will only ever pass the unqualified column names
* add quotes around table and column name, as per the specific database's rules (e.g. backticks for MySQL)

Could you push another commit for taking care of this? I think that's the last piece missing for moving it over the finish line. Thx!

Reply:
@gunnarmorling This method for MySql would be summed up as 
```
    protected String resolveColumnName(TableId tableId, String columnName) {
        StringBuilder sb = new StringBuilder();
        sb.append("`").append(tableId.table()).append("`")
                .append(".`").append(columnName).append("`");
        return sb.toString();
    }
```
Right?

Reply:
Yes, pretty much. Thinking about it though, you could even add a method `quotedColumnIdString(TableId tableId, String columnName)` to `JdbcConnection` (there already is `quotedTableIdString()`) instead, and git rid of `resolveColumnName()` again. This would take care of everything, and the constructors of the specific sub-classes of `JdbcConnection` solely would have to pass the right quote character when invoking the super-constructor. WDYT?